### DISCUSSION TITLE
Question about Provider and `alchemyId`

### DISCUSSION THREAD

**Discussion by faces-of-eth** [wevm/wagmi#701] [Q&A]
Hey fam, have a question about setup that I'm a bit confused about. In the provided documentation, this is how you're supposed to setup a provider for Alchemy.

```
import { chain, configureChains } from 'wagmi'
import { alchemyProvider } from 'wagmi/providers/alchemy'

const alchemyId = process.env.ALCHEMY_ID

const { chains, provider } = configureChains(
  [chain.mainnet, chain.polygon],
  [alchemyProvider({ alchemyId })],
)
```

1.  What is the `alchemyId`, and where can I access it? Is it actually the "API Key"?
2. This example is showing off grabbing the value from `process.env`, which would only be a Node.js thing. Is this intended to run on the client side?
3. If this is public, and you are providing the Alchemy API key, what's stopping someone from using your endpoint to run your bill up? I suppose only setting allowed domains in your configuration?

Thanks for the clarity in advance.

**Comment by tmm** [comment]
1. `alchemyId` is the API key for an app you set up on your Alchemy dashboard.
2. The example uses `process.env` to demonstrate it being an environment variable, but you should use whatever works for your framework (e.g. `process.env.NEXT_PUBLIC_`).
3. Correct, nothing is stopping anyone from using the API key unless you set up allowed domains or some other set up (e.g. set up proxy RPC service, rotate key, etc.).

**Reply by faces-of-eth** [reply]
Thanks for the quick response! It seems as if the variable is named incorrectly, or in a confusing manner. Is this something the team would accept a PR about?

Additionally, would it make sense to document the nuance expressed in points 2 & 3 in the docs?

**Reply by subject026** [reply]
+1 for `alchemyId` being a little confusing. I guessed it was the API key but wasn't 100% until I found this thread.

**Reply by tmm** [reply]
Thanks for the feedback! Updating from `xId` to `apiKey` in the 0.6 release https://github.com/wagmi-dev/wagmi/pull/760

**Comment by faces-of-eth** [comment]
Thank you for listening!
